### ISSUE TITLE
BUG: bcolz dataframes dropping columns after groupby

### ISSUE THREAD

**Issue by thequackdaddy** [dask/dask#1207]
Hello,

When I run `dask=0.8.1` with `bcolz=1.0.0`, I can get this...

```python
Python 3.5.1 |Anaconda custom (64-bit)| (default, Feb 16 2016, 09:49:46) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.2.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
%guiref   -> A brief reference about the graphical user interface.

In [1]: import dask
   ...: import dask.dataframe as ddf
   ...: import pandas as pd
   ...: import bcolz
   ...: 
   ...: data = pd.DataFrame({'a': ['a', 'a', 'b', 'b', 'c'],
   ...:                      'b': [1, 2, 1, 3, 4],
   ...:                   'c': ['z', 'x', 'x', 'z', 'x']})
   ...: 
   ...: data = bcolz.ctable.fromdataframe(data)
   ...: 
   ...: data = ddf.from_bcolz(data, chunksize=2)
   ...: 
   ...: analysis = data[['a']]
   ...: analysis = analysis.assign(Two=data['b'], One=1)
   ...: analysis.groupby('a').sum().compute()
Out[1]: 
      One  Two
a             
b'a'  2.0  3.0
b'b'  2.0  4.0
b'c'  1.0  4.0

In [2]: bcolz.__version__
Out[2]: '1.0.0'

In [3]: dask.__version__
Out[3]: '0.8.1'
```

When I upgrade to `dask=0.9.0`, the column `Two` drops. (This occurs on `bcolz` only... `pandas` seems to work. Haven't dried any other formats).

```python
Python 3.5.1 |Anaconda custom (64-bit)| (default, Feb 16 2016, 09:49:46) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.2.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
%guiref   -> A brief reference about the graphical user interface.

import dask
import dask.dataframe as ddf
import pandas as pd
import bcolz

data = pd.DataFrame({'a': ['a', 'a', 'b', 'b', 'c'],
                     'b': [1, 2, 1, 3, 4],
                  'c': ['z', 'x', 'x', 'z', 'x']})

data = bcolz.ctable.fromdataframe(data)

data = ddf.from_bcolz(data, chunksize=2)

analysis = data[['a']]
analysis = analysis.assign(Two=data['b'], One=1)
analysis.groupby('a').sum().compute()
Out[1]: 
      One
a        
b'a'  2.0
b'b'  2.0
b'c'  1.0

bcolz.__version__
Out[2]: '1.0.0'

dask.__version__

Out[3]: '0.9.0'
```

**Comment by jcrist** [comment]
The issue was an improperly inferred type. Fix coming soon.

**Comment by jcrist** [comment]
Should be fixed by #1208.

**Comment by thequackdaddy** [comment]
Thank you @jcrist
